lib/seadsa/SpecGraphInfo.cc

```diff
@@ -0,0 +1,83 @@
+
+#include "llvm/ADT/Twine.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IRReader/IRReader.h"
+#include "llvm/Pass.h"
+#include "llvm/PassRegistry.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Registry.h"
+
+#include "seadsa/Graph.hh"
+#include "seadsa/InitializePasses.hh"
+#include "seadsa/SpecGraphInfo.hh"
+#include "seadsa/support/Debug.h"
+#include <iostream>
+
+namespace seadsa {
+std::string SpecFile;
+}
+
+static llvm::cl::opt<std::string, true>
+    XSpecFile("sea-dsa-specfile", llvm::cl::desc("<Input spec bitcode file>"),
+              llvm::cl::Optional, llvm::cl::location(seadsa::SpecFile),
+              llvm::cl::init(""), llvm::cl::value_desc("filename"));
+
+namespace seadsa {
+
+void SpecGraphInfo::getAnalysisUsage(llvm::AnalysisUsage &AU) const {
+  AU.addRequired<seadsa::AllocWrapInfo>();
+  AU.addRequired<llvm::TargetLibraryInfoWrapperPass>();
+  AU.setPreservesAll();
+}
+
+void SpecGraphInfo::initialize() const {
+
+  using namespace llvm::sys::fs;
+  using namespace llvm;
+
+  m_awi = &getAnalysis<AllocWrapInfo>();
+  m_tliWrapper = &getAnalysis<TargetLibraryInfoWrapperPass>();
+
+  if (SpecFile.empty()) return;
```

Comment:
should do this check before getting analysis since they might be computationally expensive.

Reply:
Shall I also make a `isInitialized` flag to return if the SpecGraphInfo pass is already initialized?

Reply:
sounds good